### PR TITLE
cmd: write expanded form of enr in charon enr

### PR DESCRIPTION
**Pull request by xenowits**
- `charon create enr` now errors when `charon-enr-private-key` already exists.
- `charon enr` now also outputs`secp256k1 pubkey`, `signature` and `seq` in addition to the `enr` when the `--verbose` flag is provided.

category: refactor
ticket: #893

### CODE REVIEW

File: cmd/enr.go [hunk 1]
```diff
@@ -36,7 +39,7 @@ func newEnrCmd(runFunc func(io.Writer, p2p.Config, string) error) *cobra.Command
 	cmd := &cobra.Command{
 		Use:   "enr",
 		Short: "Print this client's Ethereum Node Record",
-		Long:  `Return information on this node's Ethereum Node Record (ENR)`,
+		Long:  `Prints a newly generated Ethereum Node Record (ENR) from this node's charon-enr-private-key`,
```
Comment (xenowits) [inline cmd/enr.go:42]: will add a `--silent` flag shortly which will suppress the expanded ENR output.

File: cmd/createenr.go [hunk 2]
```diff
@@ -48,7 +48,13 @@ func newCreateEnrCmd(runFunc func(io.Writer, p2p.Config, string) error) *cobra.C
 }
 
 // runCreateEnrCmd stores a new charon-enr-private-key to disk and prints the ENR for the provided config.
+// It returns an error if the key already exists.
 func runCreateEnrCmd(w io.Writer, config p2p.Config, dataDir string) error {
+	_, err := p2p.LoadPrivKey(dataDir)
+	if err == nil {
+		return errors.New("charon-enr-private-key already exists")
```
Comment (corverroos) [inline cmd/createenr.go:55]: add `z.String("path", p2p.KeyFile(datadir))`

File: cmd/enr.go [hunk 3]
```diff
@@ -64,7 +70,46 @@ func runNewENR(w io.Writer, config p2p.Config, dataDir string) error {
...
+	_, _ = sb.WriteString(fmt.Sprintf("signature: %#x\n", sig))
+	_, _ = sb.WriteString(fmt.Sprintf("seq: %d\n", seq))
+	_, _ = sb.WriteString("********************************************************************************************************\n")
+	_, _ = sb.WriteString("\n")
+
+	_, _ = w.Write([]byte(sb.String()))
+}
+
+// pubkeyBytes returns compressed public key bytes.
+func pubkeyBytes(pub *ecdsa.PublicKey) []byte {
```
Comment (corverroos) [inline cmd/enr.go]: suggest:
```
func pubkeyHex(pubkey ecdsa.PubKey) string, error {
  b, err := go_ethereum_crypto.CompressPubkey(&pubkey)
  if err ! ....
  
  return strings.Sprint(%#x, b) 
}
```

File: cmd/enr.go [hunk 4]
```diff
@@ -16,41 +16,47 @@
...
 		config  p2p.Config
 		dataDir string
+		silent  bool
 	)
 
 	cmd := &cobra.Command{
 		Use:   "enr",
-		Short: "Print this client's Ethereum Node Record",
-		Long:  `Return information on this node's Ethereum Node Record (ENR)`,
+		Short: "Print this node's Ethereum Node Record",
```
Comment (corverroos) [inline cmd/enr.go:43]: Prints a new ENR for this node

**Review by corverroos — APPROVED** [review]

File: cmd/enr.go (hunk 3 above)
Comment (xenowits) [reply cmd/enr.go]: gotcha. Here's the revised version:
```
// pubkeyHex returns compressed public key bytes.
func pubkeyHex(pubkey ecdsa.PublicKey) string {
	b := crypto.CompressPubkey(&pubkey)

	return fmt.Sprintf("%#x", b)
}
```

**Pull request merged**